pkg/config/config.go

```diff
@@ -1054,6 +1056,20 @@ type DockerManifest struct {
 	Retry          Retry    `yaml:"retry,omitempty" json:"retry,omitempty"`
 }
 
+// DockerV2 is the new Docker build pipe options.
+type DockerV2 struct {
```

Comment:
Any reason for not exposing `build_flag_templates` ?

Use-cases are `--build-context`, `--cache-from`, `--cache-to`, `--no-cache`.

Not sure if it worth to expose each single CLI options. Something like `extra_args` for advanced usage would be great.